src/services/eventService.js

```diff
@@ -0,0 +1,70 @@
+const jsonLogic = require('json-logic-js');
+const notificationService = require('./notificationService')
+
+/**
+ * Checks if string is valid json
+ * @param {string} str
+ * @returns {boolean}
+ */
+function isJson(str) {
+  try {
+    JSON.parse(str);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+/**
+ * Publish an event
+ * This method checks if there is any ruleset available for the published event
+ *
+ * @param {object} db
+ * @param {int} siteId
+ * @param {object} ruleSetData
+ * @returns {Promise<void>}
+ */
+const publish = async (db, siteId, ruleSetData) => {
+  const ruleSets = await db.NotificationRuleSet
+    .scope('includeTemplate', 'includeRecipients')
+    .findAll({where: { siteId, active: 1}})
+
+  ruleSets.forEach((ruleset) => {
+    const rulesetString = ruleset.body;
+
+    if(!isJson(rulesetString)){
+      return false;
+    }
+
+    if (jsonLogic.apply(JSON.parse(rulesetString), ruleSetData)) {
+      const { notification_template, notification_recipients } = ruleset;
+
+      const recipients = notification_recipients.map(recipient => {
+        const user = {}
+        if (recipient.emailType === 'field') {
+          user.email = recipient.value.split('.').reduce((o,i)=>o[i], ruleSetData.instance)
+        }
```

Comment:
Ik snap niet helemaal wat deze regel nou precies doet. Is dit om die `extraData.` en `submittedData.` nog uit het e-mail veld te halen?

Reply:
Ja klopt, zal er even comment plaatsen. Hij haalt het email adres uit de resource data. 
Kan dus `submittedData.email` zijn inderdaad.